### PR TITLE
fix sometimes _updateVisibleRows will not work

### PR DESCRIPTION
**Pull request by zhang740**
When some row's height is 0, should compute next row, not break the loop.
If 'break', when scroll to below the height is 0 row, `onChangeVisibleRows` will not work.
Sometimes the frames data from native report maybe incorrect.

### CODE REVIEW

File: Libraries/CustomComponents/ListView/ListView.js [hunk 1]
```diff
@@ -612,13 +615,13 @@ var ListView = React.createClass({
...
-          break;
+          continue;
         }
         var rowVisible = visibleSection[rowID];
         var min = isVertical ? frame.y : frame.x;
         var max = min + (isVertical ? frame.height : frame.width);
         if ((!min && !max) || (min === max)) {
-          break;
+          continue;
         }
```
Comment (leeight) [inline Libraries/CustomComponents/ListView/ListView.js]: 记得这个 `break` 当时是某个提交里面刻意加上的

**Comment by mkonicek** [comment]
Thanks for the pull request! @majak is the best person to review but on vacation for about a week, let's wait for his review.

**Comment by arilitan** [comment]
@zhang740 thanks for making this PR, this just solved a nasty bug for me! will have to fork this component until its merged into master

File: Libraries/CustomComponents/ListView/ListView.js (hunk 1 above)
Comment (zhang740) [inline Libraries/CustomComponents/ListView/ListView.js]: 加这个可以减少无用的计算，然而break会导致问题row（height = 0等情况）后面的row不会进行可视计算

**Comment by facebook-github-bot-2** [comment]
@zhang740 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9696/files/3b48f204d82bf244d6407f4ab589087b3daa18bc..5d38c908ab676e4fc81c2e1b01f8732974f31d9f)

**Comment by facebook-github-bot-0** [comment]
@zhang740 updated the pull request - [view changes](https://github.com/facebook/react-native/pull/9696/files/5d38c908ab676e4fc81c2e1b01f8732974f31d9f..02c53d5d64f843d9e1ea23aeb98305ff61858273)

File: Libraries/CustomComponents/ListView/ListView.js [hunk 2]
```diff
@@ -577,7 +577,10 @@ var ListView = React.createClass({
     }
     if (updatedFrames) {
       updatedFrames.forEach((newFrame) => {
-        this._childFrames[newFrame.index] = merge(newFrame);
+        // Sometimes x/y/width/heigh maybe return zero
+        if (newFrame.x != 0 || newFrame.y != 0 || newFrame.width != 0 || newFrame.height != 0) {
```
Comment (majak) [inline Libraries/CustomComponents/ListView/ListView.js]: Looks like this would exclude frames that have one of their coordinate as 0. That doesn't sound right.
Could you check only for zero size?

Comment (majak) [inline Libraries/CustomComponents/ListView/ListView.js]: Please do what @eslint-bot suggests :)

File: Libraries/CustomComponents/ListView/ListView.js [hunk 3]
```diff
@@ -612,13 +615,13 @@ var ListView = React.createClass({
           totalIndex++;
         }
         if (!frame) {
-          break;
+          continue;
```
Comment (majak) [inline Libraries/CustomComponents/ListView/ListView.js]: Are these changes still necessary, since you only add nonzero frames now?

File: Libraries/CustomComponents/ListView/ListView.js (hunk 2 above)
Comment (zhang740) [inline Libraries/CustomComponents/ListView/ListView.js]: If coordinate as 0 and size is 0 too,it will be excluded.
The reason for this change is sometimes some frame‘s data maybe lost or frame's size & coordinate is overridden by 0.
So it's possible, the frame's data width & height = 0 for coordinate as 0,but it is an error data.

File: Libraries/CustomComponents/ListView/ListView.js [hunk 4]
```diff
@@ -577,7 +577,10 @@ var ListView = React.createClass({
     }
     if (updatedFrames) {
       updatedFrames.forEach((newFrame) => {
-        this._childFrames[newFrame.index] = merge(newFrame);
+        // Sometimes x/y/width/heigh maybe return zero
```
Comment (lacker) [inline Libraries/CustomComponents/ListView/ListView.js]: heigh -> height

File: Libraries/CustomComponents/ListView/ListView.js (hunk 3 above)
Comment (zhang740) [inline Libraries/CustomComponents/ListView/ListView.js]: Sometimes as scroll fast, the frame data maybe lost, it will break the loop.

**Comment by hramos** [comment]
The CLA needs to be signed.

**Comment by lacker** [comment]
Could you add a test to cover this behavior? I am afraid it is likely to break again.

**Comment by hramos** [comment]
ping

**Comment by lacker** [comment]
It looks like this pull request has gone stale, so I am going to close it. Please feel free to reopen if you intend to keep working on this!
